internal/controller/cloudsyncmgr/cloudsync.go

```diff
@@ -65,37 +65,50 @@ func (c *CloudSyncImpl) InitiateCloudSync(isCloudSet string) (err error) {
 		if strings.Compare(strings.ToLower(isCloudSet), "true") == 0 {
 			log.Println("CloudSync init set")
 			isCloudSyncSet = true
+			//Intialize the client and hashmap storing client data
+			mqttmgr.SetClient(nil)
+			mqttmgr.InitClientData()
 		}
 	}
 	return nil
 }
 
-//StartCloudSync is used to start the sync by connecting to the broker
-func (c *CloudSyncImpl) StartCloudSync(host string) (err error) {
-	if isCloudSyncSet && len(host) > 0 {
-		log.Info("Starting the CLoudsync Mgr")
-		go mqttmgr.StartMQTTClient(host)
-	}
-	return
-}
-
 // RequestCloudSyncConf is  configuration request handler
-func (c *CloudSyncImpl) RequestCloudSyncConf(message mqttmgr.Message, topic string, clientID string) string {
+func (c *CloudSyncImpl) RequestCloudSyncConf(host string, clientID string, message mqttmgr.Message, topic string) string {
 	log.Info(logPrefix, "Publishing the data to the cloud")
```

Comment:
What about checking `isCloudSyncSet` here? If `false`, return a message regarding `cloudsync is not active'.

Reply:
Done!